control-plane/grpc/src/operations/pool/traits.rs

```diff
@@ -202,9 +203,13 @@ impl From<Pools> for pool::Pools {
     }
 }
 
-impl From<get_pools_request::Filter> for Filter {
-    fn from(filter: get_pools_request::Filter) -> Self {
-        match filter {
+impl TryFrom<get_pools_request::Filter> for Filter {
+    type Error = ReplyError;
+    fn try_from(filter: get_pools_request::Filter) -> Result<Self, Self::Error> {
+        Ok(match filter {
+            get_pools_request::Filter::Common(common_filter) => Filter::Volume(VolumeId::try_from(
```

Comment:
hmm this should be Filter::Common as well?
Should we make the change now or later? Wdyt @Abhinandan-Purkait ?

Reply:
keeping it for now, based on the input can change in another PR.